src/compas_timber/connections/butt_joint.py

```diff
@@ -0,0 +1,231 @@
+            dir_pts.append(top_min)
+
+            top_max = Point(*closest_point_on_line(max_pt, top_line))
+            bottom_min = Point(*closest_point_on_line(min_pt, bottom_line))
+            bottom_max = Point(*closest_point_on_line(max_pt, bottom_line))
+            vertices.extend([Point(*top_min), Point(*top_max), Point(*bottom_max), Point(*bottom_min)])
+
+        center = (vertices[0] + vertices[1] + vertices[2] + vertices[3]) * 0.25
+
+        angle = angle_vectors_signed(
+            subtract_vectors(vertices[0], center), subtract_vectors(vertices[1], center), sides[0].zaxis
+        )
+        if angle > 0:
+            ph = Polyhedron(
+                vertices, [[0, 1, 2, 3], [1, 0, 4, 5], [2, 1, 5, 6], [3, 2, 6, 7], [0, 3, 7, 4], [7, 6, 5, 4]]
+            )
+        else:
+            ph = Polyhedron(
+                vertices, [[3, 2, 1, 0], [5, 4, 0, 1], [6, 5, 1, 2], [7, 6, 2, 3], [4, 7, 3, 0], [4, 5, 6, 7]]
+            )
+
+        return ph
+
+    def add_features(self):
+        """Adds the required extension and trimming features to both beams.
+
+        This method is automatically called when joint is created by the call to `Joint.create()`.
+
+        """
+        assert self.main_beam and self.cross_beam  # should never happen
+        if self.features:
+            self.main_beam.remove_features(self.features)
+        start_main, start_cross = None, None
+
+        try:
+            main_cutting_plane = self.get_main_cutting_plane()[0]
+            cross_cutting_plane = self.get_cross_cutting_plane()
+            start_main, end_main = self.main_beam.extension_to_plane(main_cutting_plane)
+            start_cross, end_cross = self.cross_beam.extension_to_plane(cross_cutting_plane)
+        except BeamJoinningError as be:
+            raise be
+        except AttributeError as ae:
+            # I want here just the plane that caused the error
+            geometries = [cross_cutting_plane] if start_main is not None else [main_cutting_plane]
+            raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ae), debug_geometries=geometries)
+        except Exception as ex:
+            raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ex))
+
+        extension_tolerance = 0.01  # TODO: this should be proportional to the unit used
+
+        if self.modify_cross:
+            self.cross_beam.add_blank_extension(
+                start_cross + extension_tolerance, end_cross + extension_tolerance, self.key
+            )
+            f_cross = CutFeature(cross_cutting_plane)
+            self.cross_beam.add_features(f_cross)
+            self.features.append(f_cross)
+
+        self.main_beam.add_blank_extension(start_main + extension_tolerance, end_main + extension_tolerance, self.key)
+
+        f_main = CutFeature(main_cutting_plane)
+        self.cross_beam.add_features(MillVolume(self.subtraction_volume()))
```

Comment:
the call to `subtraction_volume()` should be protected by the `try..catch` above so that if it fails to calculate the negative volume the reason and the related debug geometries can be transmitted to the debug system via a `BeamJoinningError`.